### PR TITLE
feat: Added Restack to Installation Guides

### PR DESCRIPTION
**Pull request by jnikhila**
- Added the required files from #558 feat: Add restack as a way to deploy Appsmith
- Fixed linter issues
- Added troubleshooting and next steps section

### CODE REVIEW

**Review by sharat87 — APPROVED** [review]

File: website/docs/getting-started/setup/installation-guides/restack.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+ You can deploy Appsmith on the Restack platform with just one click.
+
+ The easiest way to do this is
+
+ * [Login to Restack](https://restack.io/login) with your GitHub or email.
+ * Connect your AWS account to create a cluster. 
+ * Search for Appsmith in the restack console and click to trigger the deployment. 
+
+You could also follow the [detailed guide for deploying Appsmith available on Restack docs.](https://www.restack.io/docs/deploy-appsmith-on-restack#overview)
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/restack.md:18]: Let's not hyperlink so much text. Just the "guide" would suffice

And maybe rephrase it as "Follow the detailed [guide](https://www.restack.io/docs/deploy-appsmith-on-restack#overview) for deploying Appsmith available on Restack docs."

Comment (danciaclara) [reply website/docs/getting-started/setup/installation-guides/restack.md:18]: The link to the guide also does not work. Let me check with Restack

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/installation-guides/restack.md (hunk 1 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/restack.md:18]: I've raised that the link is not working on the PR raised by Restack.

File: website/docs/getting-started/setup/installation-guides/restack.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+ You can deploy Appsmith on the Restack platform with just one click.
+
+ The easiest way to do this is
+
+ * [Login to Restack](https://restack.io/login) with your GitHub or email.
+ * Connect your AWS account to create a cluster. 
+ * Search for Appsmith in the restack console and click to trigger the deployment. 
+
+You could also follow the detailed guide for [deploying Appsmith](https://www.restack.io/docs/deploy-appsmith-on-restack#overview) available on Restack docs.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/restack.md:18]: @jnikhila We can add this active link - https://www.restack.io/docs/deploy-appsmith-on-kubernetes and close the PR

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/installation-guides/restack.md (hunk 2 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/restack.md]: Done.

**Review by danciaclara — APPROVED** [review]
